bin/varnishd/cache/cache_ban_lurker.c

```diff
@@ -149,13 +150,15 @@ ban_lurker_getfirst(struct vsl_log *vsl, struct ban *bt)
 		} else if (oc == &oc_mark_end) {
 			assert(move_oc == 0);
 
-			/* hold off to give lookup a chance and reiterate */
-			VSC_C_main->bans_lurker_contention++;
-			Lck_Unlock(&ban_mtx);
-			VSL_Flush(vsl, 0);
-			VTIM_sleep(cache_param->ban_lurker_holdoff);
-			Lck_Lock(&ban_mtx);
-
+			if (contested) {
```

Comment:
I do not understand why we would need this code. In the existing code, the holdoff only happens if there are ocs inbetween the two markers, and cleaning out dying and failed ocs does not change any of that.

Reply:
An object might be contested on the first pass, then moved to in between the markers, but on the second time it is dying and the *only* object between the markers. Then we do not want to *holdoff* sleep, since there were actually no contested objects between the markers. This is tricky stuff, and I might be wrong.

Reply:
ok, I think I see the intention now, but I don't think it would work like this:

We want the holdoff to happen if there were any contested ocs. Yet if one contested oc was returned and _getfirst gets called the second time, contested will be 0 and we won't hold off.
Also I would consider the intention a micro-optimization not worth the complication.